### ISSUE TITLE
Allow default servicePath "/"

### ISSUE THREAD

**Issue by kurt-s** [vfarcic/docker-flow-proxy#149]
I got a bunch of services written in Go plus a web portal (hosted by an nginx container) running in parallel using the swarm flow proxy. To make it work, I had to provide a dedicated servicePatch param for each service (which makes sense):
1. /portal
2. /search
3. /assets
etc.

The challenge I'm facing is that I'd like to load the portal (user interface written in extjs) when a user opens the browser on the host name/ip without needing to provide the additional "/portal" to the URL. But if I configure my service to use the "/" URL to route to my portal container, it, of course, now tries to route all requests to the portal which won't work:

docker service create --name portal `
  --network proxy `
  -l com.df.notify=true `
  -l com.df.servicePath=/ `
  -l com.df.port=9000 `
  kurts/at_portal

Is there a way to configure some sort of default service that is routed to when no other route matches? Another way to implement this would be to treat the "/" service path as a special one and not consider it as a catchall for every request.
Thanks
Kurt

**Comment by vfarcic** [comment]
ACLs (queries based on path, domain, and other params) are evaluated in alphabetic order by `aclName`. If not specified, `serviceName` is used instead. Proxy goes through ACLs, one at the time and forwards a request to the first one that is a match. If you specify `aclName` for the `portal` service to be, let's say, `xportal`, it will be evaluated after all other ACLs and your users will be redirected to it only if no other ACL is a match.

Ordering might be confusing so I'll use this issue as a request to add a clearer mechanism to specify a service that is evaluated last. In the meantime, please try the suggestion with `aclName` and let me know whether it fixed your problem. You'll find info about the param in the [reconfigure](https://proxy.dockerflow.com/usage/#reconfigure) section of the documentation.

**Comment by kurt-s** [comment]
thanks for the super-quick response. I could not find an example how to use the aclName param. From what I understand, if I simply configure my portal service like this

docker service create --name portal --network proxy
-l com.df.notify=true -l com.df.aclName=/xportal

the proxy will only route requests to `http://<host>/xportal` to the portal service, correct? What I really need is the proxy to forward requests to `http://<host>` to the portal service and still forward request to `http://<host>/search` to the search service.

**Comment by vfarcic** [comment]
`aclName` is not used to decide where to forward requests. It's used only to order ACLs in proxy config.

Let's say that there are two services:

```
docker service create --name some-service --network proxy ...

docker service create --name portal --network proxy ...
```

When a request is issued, proxy will evaluate `servicePath` of `portal` before `some-service` since  ACLs are ordered alphabetically. If `portal` has `servicePath` set to `/`, that path matches everything so `some-service` is never used.

Now, consider the following example:

```
docker service create --name some-service --network proxy ...

docker service create --name portal --network proxy -l com.df.aclName=xportal ...
```

This time, when a request is issued, proxy will evaluate `servicePath` of `some-service` before `xportal` (alphabetically ordered by `aclName` if specified). In this scenario, path `/` is the last so it will be used only if `some-service` does not match request. Since `some-service` is more specific, it should work correctly.

In other words, if you keep your `portal` definition as it is and only add `-l com.df.aclName=xportal`, everything should work.

As I said before, I'll make it more intuitive soon. Until then, please let me know if I managed to explain how it work. If I didn't, maybe you can join http://slack.devops20toolkit.com/. We can have a live chat over there.

**Comment by kurt-s** [comment]
works perfectly - exactly what I was looking for! 

As suggested, I'm using this config now

```
docker service create --name portal `
  --network proxy `
  -l com.df.notify=true `
  -l com.df.servicePath=/ `
  -l com.df.aclName=xportal `
  -l com.df.port=9000 `
  kurts/at_portal

```
All other servcies don't have the aclName parameter set and work based on the servicePath.
Thank you very much for the quick help!

**Comment by vfarcic** [comment]
I'm glad it worked out. I just created a new issue #151 that should make this simpler.
